db.go

```diff
@@ -0,0 +1,452 @@
+package rendezvous
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"database/sql"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3"
```

Comment:
probably good to have the client and server code separated out. I'd rather not import sqlite in my program if i don't have to

Reply:
Probably a good idea. 

Should I make a separate `go-libp2p-rendezvous-service` package?
We might be able to get away with simply putting it in a subpackage though, so that unless you import `github.com/go-libp2p-rendezvous/svc` (or however we call it), you shouldn't get the db code.

Reply:
yeah, subpackage is fine for now i think

Reply:
ok, will restructure.

Reply:
Restructured in https://github.com/libp2p/go-libp2p-rendezvous/pull/1/commits/c487c2077b1528816577f043c1f56d143915a63b

We now have a db subpackage with the interface, and implementations of the db in subpackages of that.

So now there is no dependency of the main package (or the dbi interface package) on SQLite or any particular implementation, and clients can freely import the client interface without needing to link SQLite code.